js/taskOverlay/taskOverlay.js

```diff
@@ -89,7 +120,27 @@ window.taskOverlay = {
 
     // show the task elements
     tasks.forEach(function (task, index) {
-      const el = createTaskContainer(task, index)
+      const el = createTaskContainer(task, index, {
+        tabSelect: function () {
+          browserUI.switchToTask(task.id)
+          browserUI.switchToTab(this.getAttribute('data-tab'))
+
+          taskOverlay.hide()
+        },
+        tabDelete: function (item) {
+          var tabId = item.getAttribute('data-tab')
+
+          browserUI.destroyTab(tabId)
+
+          // if there are no tabs left, remove the task
+          if (task.tabs.count() === 0) {
+            // remove the task element from the overlay
+            getTaskContainer(task.id).remove()
+            // close the task
+            browserUI.closeTask(task.id)
+          }
+        }
+      })
```

Comment:
Previously, there was a loop where taskOverlay depended on taskOverlayBuilder, and taskOverlayBuilder depended on taskOverlay, which it isn't possible to do any more. I fixed this by passing functions into the builder instead.